### ISSUE TITLE
Make Manila integration optional

### ISSUE THREAD

**Issue by berendt** [vexxhost/magnum-cluster-api#224]
Currently, a manila endpoint must be present. We don't have a manila service at our end and would actually be reluctant to add it. Would it be possible to make the integration of Manila configurable (opt-out)?

**Comment by mnaser** [comment]
@berendt I don't believe that it's required at the moment, are you seeing issues where it's giving an error that it's required?   I believe it should just log a warning and then do nothing

**Comment by berendt** [comment]
I had prepared this in my testbed where Manila is not activated by default.  Then an error came up that a manila endpoint was missing. So I think it is not currently optional.

**Comment by mnaser** [comment]
@berendt can you share more details?

https://github.com/vexxhost/magnum-cluster-api/blob/391cdacf7fd368f07f149a1c8792b15a4617843a/magnum_cluster_api/resources.py#L246-L328

This block here checks if it's enabled

https://github.com/vexxhost/magnum-cluster-api/blob/391cdacf7fd368f07f149a1c8792b15a4617843a/magnum_cluster_api/integrations/manila.py#L20-L21
https://github.com/vexxhost/magnum-cluster-api/blob/391cdacf7fd368f07f149a1c8792b15a4617843a/magnum_cluster_api/integrations/common.py#L26-L33

So it should break out...
https://github.com/vexxhost/magnum-cluster-api/blob/391cdacf7fd368f07f149a1c8792b15a4617843a/magnum_cluster_api/integrations/common.py#L26-L33

**Comment by berendt** [comment]
I'm also going to watch it on Thursday. I haven't had time to continue yet.

**Comment by okozachenko1203** [comment]
> I'm also going to watch it on Thursday. I haven't had time to continue yet.

@berendt did you have a chance to check again?

**Comment by mnaser** [comment]
@berendt 👋 did you have any other issues?
